mobify/platform-scaffold
# WEB-1531: Add Meta Data to Pages (SEO)

Add SEO helpful `<meta>` and `<title>` tags to the site header via the [`react-helmet`](https://www.npmjs.com/package/react-helmet) dependency.

**JIRA**: [WEB-1531](https://mobify.atlassian.net/browse/WEB-1531)

## Changes
- Add new dependency: `react-helmet`
- Use `react-helmet` to add `<title>`, `<meta name="description">`, `<meta name="keyword">` tags to the site header
- New `pageMeta` key added to the `categories` and `products` redux stores
    - Key is formatted as follows...
        ```
        pageMeta: {
            title: '...',
            description: '...',
            keyword: '...',
        }
        ```
- The `pageMeta` key, as used in a container with `react-helmet`, appears as follows...
    ```
    <Helmet>
        <title>{pageMeta.title}</title>
        <meta name="description" content={pageMeta.description} />
        <meta name="keywords" content={pageMeta.keywords} />
    </Helmet>
    ```
- Containers affected:
    - Product Details pages
    - Product List pages

## Todos
- ~~(if applicable) analytics events instrumented to measure impact of change~~

## How to test-drive this PR
- **IMPORTANT:** run `npm i` to install new dependency
- Run `npm start`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Visit any of the affected pages (as listed above)
- **Verify** that the site `title` updates appropriately
- **Verify** that a `<meta name="description">` tag is present on that pages and matches what's on desktop
- **Verify** that a `<meta name="keywords">` tag is present on that pages and matches what's on desktop (in the case of Merlin's it will be empty)
- _Repeat above steps with the SFCC connector_
